src/core/ext/transport/chttp2/transport/parsing.cc

```diff
@@ -409,67 +409,75 @@ static void on_initial_header(void* tp, grpc_mdelem md) {
     gpr_free(value);
   }
 
-  if (grpc_slice_eq(GRPC_MDKEY(md), GRPC_MDSTR_GRPC_STATUS) &&
-      !grpc_mdelem_eq(md, GRPC_MDELEM_GRPC_STATUS_0)) {
-    /* TODO(ctiller): check for a status like " 0" */
-    s->seen_error = true;
-  }
+  if (GRPC_MDELEM_STORAGE(md) == GRPC_MDELEM_STORAGE_STATIC) {
+    if (md.payload == GRPC_MDELEM_GRPC_STATUS_1.payload ||
```

Comment:
The reason I don't use grpc_mdelem_eq is because if the payload is not equal, I already know that the metadata elements are not equal because I confirmed that md is static. If I used grpc_mdelem_eq here, then if the payloads are not equal, we still execute more instructions to determine if they're equal or not. I'm avoiding the execution of these additional instructions.

Reply:
I see. Could you add a comment here too? Thanks.

Reply:
Added a comment in each place.